src/app/functions/server/getReferrerUrl.ts

```diff
@@ -0,0 +1,32 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import type { headers as headersGetter } from "next/headers";
+
+type ReferrerUrlProps = {
+  headers: ReturnType<typeof headersGetter>;
+  referrerParam: string | undefined;
+};
+
+export function getReferrerUrl({
+  headers,
+  referrerParam,
+}: ReferrerUrlProps): string {
+  const referrer = headers.get("referer");
+  const serverUrl = process.env.SERVER_URL as string;
+  const isRouteWithinMonitor = referrer && referrer.includes(serverUrl);
+
+  if (isRouteWithinMonitor) {
+    return referrer.replace(serverUrl, "");
+  }
+
+  switch (referrerParam) {
+    case "dashboard":
+      return "/redesign/user/dashboard";
+    case "fix":
+      return "/redesign/user/dashboard/fix/data-broker-profiles/start-free-scan";
+    default:
+      return "";
```

Comment:
Should we consider making this return `null`, and hiding the back button if it is?

Reply:
I had it go to `/redesign/user/dashboard` as a fallback before. [f25e84a](https://github.com/mozilla/blurts-server/pull/3621/commits/f25e84a93ee9e5bc3eb725d64f05e8c19b5d80a3) would now hide the “Go back” button if there is no previous route. One downside: After reloading this page there would be also a referrer present and thus we would not show the button.

Reply:
Sorry, I'm not sure I follow - shouldn't the query param still be present, and thus the button as well, when you refresh?

Reply:
Sorry, yeah: I forgot to mention I only mean the specific case where the query parameter might get stripped for some reason and it only uses the `referer` value from the header. No big deal I think — just wanted to mention it.